### DISCUSSION TITLE
Add bugchecks for un-recoverable and or fatal errors, non-terminating 'wait for..'

### DISCUSSION THREAD

**Discussion by dv-msft** [microsoft/ebpf-for-windows#2247] [Ideas]
### Describe the bug

We currently have 'asserts' for un-recoverable and/or fatal errors.  These asserts are triggered in debug builds but are compiled out in 'release' builds.

This can often lead to silent failures and/or buggy system behavior in release builds.  Also, considering that most ebpf consumers would prefer running the retail builds (ebpf program verification times are un-acceptably slow in debug builds), the subsequent inconsistent behavior would only end up confusing the ebpf end customers.

We should bugcheck the system and generate a crash dump to debug and fix issues such as:
- Un-recoverable errors
- Fatal errors
- indeterminate 'waits'

### Steps taken to reproduce bug

N/A

### Expected behavior

The system should bugcheck and generate a crash dump.

### Actual outcome

The failures are ignored and system behavior becomes indeterminate.

**Comment by shankarseal** [comment]
Need to investigate how crash dumps generated in eBPF binaries in production can be consumed for investigation.
